pkg/locker/locker.go

```diff
@@ -90,6 +93,13 @@ func (l *Locker) Lock(name string) {
 	// once locked then we can decrement the number of waiters for this lock
 	nameLock.Lock()
```

Comment:
No need to expose the condition variable as the API, only need to use it internally instead of using a literal sync.Mutex.